src/app/storage/duck/sagas.ts

```diff
@@ -85,8 +86,18 @@ function* removeStorageSaga(action) {
     );
     const migStorageResource = new MigResource(MigResourceKind.MigStorage, migMeta.namespace);
 
-    const arr = yield Promise.all([
-      client.delete(secretResource, name),
+    const secretResourceList = yield client.list(
+      secretResource,
+      getTokenSecretLabelSelector(MigResourceKind.MigStorage, name)
+    );
+
+    const secretResourceName =
+      secretResourceList.data.items && secretResourceList.data.items.length > 0
+        ? secretResourceList.data.items[0].metadata.name
+        : '';
```

Comment:
This is an odd code branch. Do we want to ask the client to delete with a resource name of `''`? It seems like this is an unexpected result.

Reply:
@eriknelson Agreed. However, I did not have a good answer to this question : What if Secret does not exist at all while we are deleting the Storage? Removing a blank resource doesn't have an adverse effect on the outcome. I agree that it looks like an odd branch. Open for better idea to handle this. I just wanted to leverage benefits of Promise.all() and make less changes to current code.